luda-adventure/src/app/game/types/game_object/types/positioner/positioner.rs

```diff
@@ -50,107 +51,116 @@ impl Positioner {
+                ) {
+                    let movement_until_collision = Movement {
+                        start_time: prediction_start_time,
+                        end_time: collision.start_time,
+                        start_position: prediction_start_position,
+                        end_position: collision.start_position,
+                        velocity: velocity_after_last_collision,
+                        movement_state: MovementState::MoveAlongAxisToCollide,
+                    };
+                    self.movement_plan
+                        .predicted_movement_list
+                        .push(movement_until_collision);
+                } else {
+                    let delta_time = last_collision.end_time - last_collision.start_time;
+                    let end_position =
+                        velocity_after_last_collision * delta_time + prediction_start_position;
+                    let movement_until_last_collision_end = Movement {
+                        start_time: prediction_start_time,
+                        end_time: last_collision.end_time,
+                        start_position: prediction_start_position,
+                        velocity: velocity_after_last_collision,
+                        end_position,
+                        movement_state: MovementState::MoveAlongAxis,
+                    };
+                    self.movement_plan
+                        .predicted_movement_list
+                        .push(movement_until_last_collision_end);
                 }
             }
-            None => {
-                let movement_until_end = Movement {
-                    start_time: prediction_start_time,
-                    end_time: self.movement_plan.directed_movement.end_time,
-                    start_position: prediction_start_position,
-                    velocity: self.movement_plan.directed_movement.velocity,
-                    end_position: self.movement_plan.directed_movement.end_position,
-                };
+            super::MovementState::MoveAlongAxisToCollide | super::MovementState::Stuck => {
+                let stay_forever =
+                    Movement::stay_forever(prediction_start_position, prediction_start_time);
                 self.movement_plan
                     .predicted_movement_list
-                    .push(movement_until_end);
+                    .push(stay_forever);
             }
-        };
+        }
     }
 
     pub fn get_position(&self, current_time: Time) -> Xy<crate::app::game::Tile> {
         self.movement_plan
             .get_position(current_time)
             .expect("get_position() of PlayerCharacter called before prediction")
     }
+
+    fn get_last_prediction(&self) -> Movement {
+        match self.movement_plan.predicted_movement_list.last() {
+            Some(movement) => movement.clone(),
+            None => Movement {
+                start_time: (f32::NEG_INFINITY).ms(),
```

Comment:
Why start_time here is -inf?

Reply:
Kind of fallback. Would it be better to just emit an error?

Reply:
At least this code should let anyone knows that you want to put this for fallback.

Reply:
No longer use this.